rotkehlchen/tasks/manager.py

```diff
@@ -397,25 +401,38 @@ def _maybe_decode_evm_transactions(self) -> None:
     def _maybe_check_premium_status(self) -> None:
         """
         Validates the premium status of the account and if the credentials are not valid
-        it deactivates the user's premium status.
+        it retries for 3 times before deactivating the user's premium status, but if the
+        credentials are valid it will reactivate the user's premium status.
         """
         now = ts_now()
         if now - self.last_premium_status_check < PREMIUM_STATUS_CHECK:
             return
 
         db_credentials = self.database.get_rotkehlchen_premium()
-        if db_credentials:
-            try:
-                premium_create_and_verify(db_credentials)
-            except PremiumAuthenticationError as e:
-                message = (
-                    f'Could not authenticate with the rotkehlchen server with '
-                    f'the API keys found in the Database. Error: {str(e)}. Will '
-                    f'deactivate the premium status.'
-                )
-                self.msg_aggregator.add_error(message)
-                self.deactivate_premium()
-        self.last_premium_status_check = now
+        if not db_credentials:
+            self.last_premium_status_check = now
+            return
+
+        try:
+            premium_create_and_verify(db_credentials)
+        except PremiumAuthenticationError as e:
+            if not self.premium_check_retries >= PREMIUM_CHECK_RETRY_LIMIT:
+                self.premium_check_retries += 1
+                self.last_premium_status_check = now
+                return
+            message = (
+                f'Could not authenticate with the rotkehlchen server with '
+                f'the API keys found in the Database. Error: {str(e)}. Will '
+                f'deactivate the premium status.'
+            )
+            self.msg_aggregator.add_error(message)
+            self.deactivate_premium()
+        else:
+            if self.premium_check_retries >= PREMIUM_CHECK_RETRY_LIMIT:
+                self.activate_premium(db_credentials)
```

Comment:
Question: Shouldn't this happen always if the premium is deactivated? I don't know why you introduce the `>=` check here

Reply:
If I don't add the check then, every time the check is run, it'll try to reactivate the premium regardless of whether it is already active or not. 

My assumption is that this task runs regardless of premium status, if that's not the case then I can remove the check.

Reply:
I was thinking in a different scenario but is not possible since we check the premium when we start the application. I think that the only possible way to have it deactivated and having to reactivate it is by having increased `self.premium_check_retries` before